packages/bff/src/services/agreementService.ts

```diff
@@ -103,6 +115,104 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return enrichAgreement(agreement, clients, ctx);
     },
+
+    async addAgreementConsumerDocument(
+      agreementId: string,
+      doc: bffApi.addAgreementConsumerDocument_Body,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<Buffer> {
+      const documentPath = `${config.consumerDocumentsPath}/${agreementId}`;
+      const documentContent = Buffer.from(await doc.doc.arrayBuffer());
+      const documentId = randomUUID();
+
+      await fileManager.storeBytes(
+        config.consumerDocumentsContainer,
+        documentPath,
+        documentId,
+        doc.doc.name,
+        documentContent,
+        logger
+      );
+
+      const seed: agreementApi.DocumentSeed = {
+        id: documentId,
+        prettyName: doc.prettyName,
+        name: doc.doc.name,
+        contentType: doc.doc.type,
+        path: documentPath,
+      };
+
+      await agreementProcessClient.addAgreementConsumerDocument(seed, {
+        params: { agreementId },
+        headers,
+      });
+
+      return documentContent;
+    },
+
+    async getAgreementConsumerDocument(
+      agreementId: string,
+      documentId: string,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<Buffer> {
+      const documentSeed =
+        await agreementProcessClient.getAgreementConsumerDocument({
+          params: { agreementId, documentId },
+          headers,
+        });
+
+      assertContentMediaType(documentSeed.contentType, agreementId, documentId);
+
+      const documentBytes = await fileManager.get(
+        config.consumerDocumentsContainer,
+        documentSeed.path,
+        logger
+      );
+
+      return Buffer.from(documentBytes);
+    },
+
+    async getAgreementContract(
+      agreementId: string,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<Buffer> {
+      logger.info(`Retrieving contract for agreement ${agreementId}`);
+
+      const agreement = await agreementProcessClient.getAgreementById({
+        params: { agreementId },
+        headers,
+      });
+
+      if (!agreement.contract) {
+        match(agreement.state).with("ACTIVE", "SUSPENDED", "ARCHIVED", () => {
+          throw contractException(agreementId);
+        });
+        throw contractNotFound(agreementId);
```

Comment:
Yes i agree with you, i tried that but sadly `otherwise` does not infer well types and typescript does not understand that the if branch is exhaustive. Maybe i can simply use an if statement here.

Reply:
You could use a second `.with` with all the other states, and then `exhaustive`. Or just an if statement also is ok :)